### PR TITLE
refactor(cm): force kill the proc that failed to response takover call

### PR DESCRIPTION
**Pull request by HJianBo**
port from: #7026

### CODE REVIEW

File: apps/emqx/src/emqx_cm.erl [hunk 1]
```diff
@@ -274,15 +282,20 @@ open_session(false, ClientInfo = #{clientid := ClientId}, ConnInfo) ->
...
+                                     ChanPid) of
+                                  {ok, Pendings} ->
+                                      Session1 = emqx_persistent_session:persist(
+                                                   ClientInfo, ConnInfo, Session),
+                                      register_channel(ClientId, Self, ConnInfo),
+                                      {ok, #{session  => Session1,
+                                             present  => true,
+                                             pendings => Pendings}};
+                                  {error, _} ->
+                                      CreateSess()
```
Comment (HJianBo) [inline apps/emqx/src/emqx_cm.erl:297]: I'm not sure, if takeover-end fails, do we need to clean up some kind of resource, such as the already persistent session?

**Review by zmstone — APPROVED** [review]

File: apps/emqx/src/emqx_cm.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx/src/emqx_cm.erl:297]: what do you mean by already persistent session ?

Comment (HJianBo) [reply apps/emqx/src/emqx_cm.erl:297]: 🤔I'm thinking if the request {takeover, 'end'} fails here, do I need to perform any cleanup action? For example, clean up the old session in the persistent store?

Comment (zmstone) [reply apps/emqx/src/emqx_cm.erl:297]: For the on-disk sessions?
Don’t worry about it for now

**Review by zmstone — APPROVED** [review]

**Pull request merged**